fenix/app/src/main/java/org/mozilla/fenix/components/metrics/Event.kt

```diff
@@ -46,5 +46,27 @@ sealed class Event {
          * Event recording the first time a URI is loaded in Firefox in a 24 hour period.
          */
         object FirstUriLoadForDay : GrowthData("ja86ek")
+
+        /**
+         * Event recording when User is "activated" in first week of usage.
+         * Activated = if the user is active 3 days in their first week and
+         *                  if they search once in the latter half of that week (days 4-7).
+         */
+        object UserActivated : GrowthData("13mh5x")
+    }
+
+    /**
+     * Events for helping track user data for growth events
+     */
+    sealed class ActivatedUserData(val tokenName: String) : Event() {
+        /**
+         * Event recording search usage during days 2-7 of first week of install
+         */
+        object FirstWeekSearchActivity : GrowthData("yeahnope")
+
+        /**
+         * Event recording app usage during days 2-7 of first week of install
+         */
+        object FirstWeekUsageActivity : GrowthData("nopeyeah")
```

Comment:
Instead of the above, you could also consider using only the `UserActivated` event, with some kind of flag to indicate where it was being triggered from. If it was a search event, you could update that state and otherwise update the usage. Then, the search state and usage state could be checked simultaneously to see if we should forward the event to Adjust. For example:

```kotlin
data class UserActivated(val fromSearch: Boolean) : GrowthData()

// shouldSend()
                Event.GrowthData.UserActivated -> {
                    currentTime.duringFirstWeek() &&
                        settings.adjustUseCount.value >= daysActivatedThreshold &&
                        settings.adjustEarlySearchCount.value >= searchesActivatedThreshold &&
                        !settings.adjustUserActivatedSent
                }

// updatePersistentState (new function, see review comment about `updateSentState`)
when (event) {
    is UserActivated -> {
         if (fromSearch) {
             // update pref about search
         } else {
            // update pref about usage
         }
    }
}
```

Reply:
Ah thank you!
I made some changes, this is a much more elegant implementation.

using `shouldTrack` as intended for checking the threshold necessary for sending the "activated" signal while taking advantage of a flag to update the local data for the usage threshold in `updatePersistentState` if the `shouldTrack` call returns `false`

I also updated the test cases to reflect this new paradigm.